doc/wallet-storage.md

```diff
@@ -0,0 +1,255 @@
+# Wallet Storage Design
+
+In current state libindy allows to plug different wallet implementations. Plugged wallet now handles both security
+and storage layers. This design proposes to restrict our plugged interface by handling only storage layer.
+All encryption will be performed in libindy. It will simplify plugged wallets and warranty good security level
+for 3d party wallets implementations.
+
+Also proposals enhances our API for efficient and flexible search with paging support.
+
+## Goals and ideas
```

Comment:
I think we need to have a diagram/doc that lists all libindy calls and how they work with the Wallet.
I think it confuses community a lot, where some methods doesn't work with wallet at all, some work with them quite expectedly, and some may have access to the wallet implicitly and surprisingly (like caches, etc.).
I think it will not only provide a better documentation and help community to understand the API, but also can check whether we always work with just ONE wallet at a time (so, we can support Case 2 from the comment above).